app/waf/src/models/as3.model.ts

```diff
@@ -1,387 +1,47 @@
-  use: string;
-  constructor(name: string) {
-    super();
-    this.use = name;
-  }
-}
-class classwafpolicy {
-  public wafpolicy: string;
-}
-export class AS3Action extends AS3Object {
-  type: string;
-  wafpolicy: object;
-  wafs: Wafpolicy[];
-
-  constructor(action: Action, params: {[key: string]: Object}) {
-    super();
-    this.type = action.type;
-    this.wafs = <Wafpolicy[]>params.wafs;
-    if (this.type === 'waf') {
-      this.wafpolicy = action.policy;
-      if (!isNullOrUndefined(this.wafpolicy)) {
-        let wafid = <classwafpolicy>this.wafpolicy;
-        for (let onewaf of this.wafs) {
-          if (onewaf.id === wafid.wafpolicy && onewaf.name) {
-            this.wafpolicy = <AS3USEPolicy>new AS3USEPolicy(onewaf.name);
-            break;
-          }
-        }
-      }
-    }
-  }
-  toJSON(): Object {
-    let obj: AS3JSONObject = {
-      type: this.type,
-    };
-    if (this.type === 'waf') {
-      if (!isNullOrUndefined(this.wafpolicy)) {
-        obj.policy = this.wafpolicy;
-      }
-    }
-    return obj;
-  }
-}
-export class AS3Rule extends AS3Object {
-  name: string;
-  conditions: AS3Condition[];
-  actions: AS3Action[];
-
-  constructor(rule: Rule, params: {[key: string]: Object}) {
-    super();
-
-    this.conditions = [];
-    this.actions = [];
-
-    let conditions = <Condition[]>rule.conditions;
-    let actions = <Action[]>rule.actions;
-
-    if (rule.name) {
-      this.name = rule.name;
-    }
-    for (let condition of conditions) {
-      this.conditions.push(new AS3Condition(condition));
-    }
-
-    for (let action of actions) {
-      this.actions.push(new AS3Action(action, params));
-    }
-  }
-}
-export class AS3EndpointPolicy extends AS3Object {
-  name?: string;
-  rules: AS3Rule[];
-  constructor(params: {[key: string]: Object}) {
-    super();
-    let endpointpolicy = <Endpointpolicy>params['endpointpolicy'];
-    let rules = <Rule[]>params.rules;
-    this.class = 'Endpoint_Policy';
-    this.name = endpointpolicy.name;
-
-    this.rules = [];
-
-    for (let rule of rules) {
-      this.rules.push(new AS3Rule(rule, params));
-    }
-  }
-  toJSON(): Object {
-    let obj: AS3JSONObject = {
-      class: this.class,
-      rules: this.rules,
+    this.declaration = {
+      class: 'ADC',
+      schemaVersion: '3.0.0',
+      id: adc.id,
     };
-    return obj;
-  }
-}
-export class AS3Pool extends AS3Object {
-  name: string;
-  monitors: string[];
-  members: AS3PoolMember[];
-  constructor(params: {[key: string]: Object}) {
-    super();
-
-    this.class = 'Pool';
-    this.monitors = ['http'];
 
-    let pool = <Pool>params.pool;
-    if (pool.name) {
-      this.name = pool.name;
+    if (adc.name) {
+      this.declaration.label = adc.name;
     }
 
-    let members = <Member[]>params.members;
-
-    this.members = [];
-    for (let member of members) {
-      this.members.push(new AS3PoolMember(member));
+    if (adc.description) {
```

Comment:
`if` can be removed: 
if `adc.description` is not null, this.declaration.remark is adc.description. 
if `adc.description` is null, this.declaration.remark would be null. 
No difference. Same to adc.name.

Reply:
I do not hope to add 'label' and 'remark' key into AS3 request body, so that I add the 'if'.

Reply:
Remove these lines, since they are useless in fact.